File: CSharp/DP.Domain/Samples/Factory/StcDbFactory.cs
using System;

namespace DP.Domain.Samples.Factory
{
    /// <summary>
    /// Static Database factory
    /// </summary>
    public class StcDbFactory
    {
        public static DbContext Create(DbEnum dbEnum)
        {
            switch (dbEnum)
            {
                case DbEnum.DataMart:
                    return new DbContext()
                    {
                        Server = "DataMart",
                        ConnectionStr = "DataMart connection string"
                    };
                case DbEnum.History:
                    return new DbContext()
                    {
                        Server = "History",
                        ConnectionStr = "History connection string"
                    };
                case DbEnum.Online:
                    return new DbContext()
                    {
                        Server = "Online",
                        ConnectionStr = "Online connection string"
                    };
                default:
                    throw new Exception("No mapping database settings!");
            }

        }

        public static DbContext CreateDataMart()
        {
            return new DbContext()
            {
                Server = "DataMart",
                ConnectionStr = "DataMart connection string"
            };
        }

        public static DbContext CreateHistory()
        {
            return new DbContext()
            {
                Server = "History",
                ConnectionStr = "History connection string"
            };
        }

        public static DbContext CreateOnline()
        {
            return new DbContext()
            {
                Server = "Online",
                ConnectionStr = "Online connection string"
            };
        }
    }


}

File: Python/Samples/Factory/StcDbFactory.py
from Factory_models import DbEnum, DbContext


class StcDbFactory(object):
    @staticmethod
    def create(dbEnum:DbEnum):
        
        return {
            DbEnum.DataMart: DbContext(server="DataMart", connectionStr="DataMart connection string"),
            DbEnum.History: DbContext(server="History", connectionStr="History connection string"),
            DbEnum.Online: DbContext(
                server="Online", connectionStr="Online connection string")
        }[dbEnum]

    @staticmethod
    def createDataMart():
        dbContext = DbContext(
            server="DataMart", connectionStr="DataMart connection string")
        return dbContext

    @staticmethod
    def createHistory():
        dbContext = DbContext(
            server="History", connectionStr="History connection string")
        return dbContext

    @staticmethod
    def createOnline():
        dbContext = DbContext(
            server="Online", connectionStr="Online connection string")
        return dbContext
